src/CSharp/CodeCracker/Style/ForInArrayCodeFixProvider.cs

```diff
@@ -50,10 +53,27 @@ where controlVarId.Equals(initSymbol)
                                      select s).ToList();
             var arrayDeclaration = arrayDeclarations.First();
             var blockForFor = forBlock.RemoveNode(arrayDeclaration, SyntaxRemoveOptions.KeepLeadingTrivia);
+
             var forEachStatement = SyntaxFactory.ForEachStatement(SyntaxFactory.ParseTypeName("var"), arrayDeclaration.Declaration.Variables.First().Identifier, arrayAccessor.Expression, blockForFor)
                 .WithLeadingTrivia(forStatement.GetLeadingTrivia())
                 .WithTrailingTrivia(forStatement.GetTrailingTrivia())
                 .WithAdditionalAnnotations(Formatter.Annotation);
+
+            var foreachVariable = SyntaxFactory.IdentifierName(forEachStatement.Identifier.Text);
+            Func<ForEachStatementSyntax, List<ElementAccessExpressionSyntax>> getElementsAccessorsToReplace;
+            getElementsAccessorsToReplace = (foreachStatement) => forEachStatement.DescendantNodes()
+                            .OfType<ElementAccessExpressionSyntax>()
+                            .Where(eae => eae.ArgumentList.Arguments.Any(a => a.Expression.ToFullString().Trim() == forVariable.Identifier.ToFullString().Trim()))
+                            .ToList();
```

Comment:
@AlbertoMonteiro This will lead to a bug when there is an accessor that is being applied to a different array. For example:

````csharp
for (var i = 0; i < array.Length; i++)
{
    var item = array[i];
    count += array[i];
    anotherCount += anotherArray[i];
}
````

This will incorrectly be changed to:

````csharp
foreach (var item in array)
{
    count += item;
    anotherCount += item;
}
````